pychunkedgraph/backend/chunkedgraph_comp.py

```diff
@@ -67,3 +104,49 @@ def get_latest_roots(cg,
         root_ids.extend(result)
 
     return np.array(root_ids, dtype=np.uint64)
+
+
+def get_delta_roots(cg,
+                    time_stamp_start: datetime.datetime,
+                    time_stamp_end: Optional[datetime.datetime] = None,
+                    min_seg_id: int = 1,
+                    n_threads: int = 1) -> Sequence[np.uint64]:
+
+    # Create filters: time and id range
+    max_seg_id = cg.get_max_seg_id(cg.root_chunk_id) + 1
+
+    n_blocks = np.min([n_threads + 1, max_seg_id-min_seg_id+1])
+    seg_id_blocks = np.linspace(min_seg_id, max_seg_id, n_blocks, dtype=np.uint64)
+
+    cg_serialized_info = cg.get_serialized_info()
+
+    if n_threads > 1:
+        del cg_serialized_info["credentials"]
+
+    multi_args = []
+    for i_id_block in range(0, len(seg_id_blocks) - 1):
+        multi_args.append([seg_id_blocks[i_id_block],
+                           seg_id_blocks[i_id_block + 1],
+                           cg_serialized_info, time_stamp_start, time_stamp_end])
+
+    # Run parallelizing
+    if n_threads == 1:
+        results = mu.multiprocess_func(_read_delta_root_rows_thread,
+                                       multi_args, n_threads=n_threads,
+                                       verbose=False, debug=n_threads == 1)
+    else:
+        results = mu.multisubprocess_func(_read_delta_root_rows_thread,
+                                          multi_args, n_threads=n_threads)
+
+    new_root_ids = []
+    expired_root_id_candidates = []
+    for r1, r2 in results:
+        new_root_ids.extend(r1)
+        expired_root_id_candidates.extend(r2)
+    expired_root_id_candidates = np.array(expired_root_id_candidates, dtype=np.uint64)
+    expired_root_id_candidates = np.unique(expired_root_id_candidates)
+    rows = cg.read_node_id_rows(node_ids=expired_root_id_candidates,
+                                end_time=time_stamp_start)  
+    expired_root_ids = [k for (k, v) in rows.items()]
```

Comment:
Why is this part necessary?

Reply:
The reason for this is to filter out the root_ids that were created after the time stamp start, by looking for all new root_ids and going backward one FormerParent, the parallelized filter includes the root_ids that were created and then expired during the time interval, this filter removes those.  I forgot I had put this in.  I realized after your note that I needed a Child column filter on this call though, and I added comments.